src/test/scala/org/broadinstitute/dsde/workbench/sam/service/UserServiceSpec.scala

```diff
@@ -203,21 +204,55 @@ class UserServiceSpec extends FlatSpec with Matchers with TestSupport with Mocki
   }
 
   /**
     * GoogleSubjectId    Email
     *      yes            skip    ---> User exists. Do nothing.
     */
   it should "return conflict when there's an existing subject for a given googleSubjectId" in{
-    val user = genCreateWorkbenchUserAPI.sample.get
+    val user = genCreateWorkbenchUser.sample.get
     dirDAO.createUser(WorkbenchUser(user.id, Some(user.googleSubjectId), user.email)).futureValue
-    val res = service.registerUser(user).failed.futureValue
-    res === new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"user ${user} already exists")) shouldBe true
+    val res = service.registerUser(user).failed.futureValue.asInstanceOf[WorkbenchExceptionWithErrorReport]
+    Eq[WorkbenchExceptionWithErrorReport].eqv(res, new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"user ${user} already exists"))) shouldBe true
+  }
+
+  "UserService inviteUser" should "create a new user" in{
+    val user = genInviteUser.sample.get
+    service.inviteUser(user).futureValue
+    val res = dirDAO.loadUser(user.inviteeId).futureValue
+    res shouldBe Some(WorkbenchUser(user.inviteeId, None, user.inviteeEmail))
+  }
+
+  it should "return conflict when there's an existing subject for a given userId" in{
+    val user = genInviteUser.sample.get
+    val email = genNonPetEmail.sample.get
+    dirDAO.createUser(WorkbenchUser(user.inviteeId, None, email)).futureValue
+    val res = service.inviteUser(user).failed.futureValue.asInstanceOf[WorkbenchExceptionWithErrorReport]
+    Eq[WorkbenchExceptionWithErrorReport].eqv(res, new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"identity with id ${user.inviteeId} already exists"))) shouldBe true
+  }
+
+  it should "return conflict when there's an existing subject for a given email" in{
+    val user = genInviteUser.sample.get
+    val userId = genWorkbenchUserId(System.currentTimeMillis())
+    dirDAO.createUser(WorkbenchUser(userId, None, user.inviteeEmail)).futureValue
+    val res = service.inviteUser(user).failed.futureValue.asInstanceOf[WorkbenchExceptionWithErrorReport]
+    Eq[WorkbenchExceptionWithErrorReport].eqv(res, new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"email ${user.inviteeEmail} already exists"))) shouldBe true
+  }
+
+  "invite user and then create user with same email" should "update googleSubjectId for this user" in {
+    val user = genCreateWorkbenchUser.sample.get
+    service.inviteUser(InviteUser(user.id, user.email)).futureValue
+    val res = dirDAO.loadUser(user.id).futureValue
+    res shouldBe Some(WorkbenchUser(user.id, None, user.email))
+
+    service.createUser(user).futureValue
+    val updated = dirDAO.loadUser(user.id).futureValue
+    updated shouldBe Some(WorkbenchUser(user.id, Some(user.googleSubjectId), user.email))
   }
```

Comment:
Pending my other comments from above regarding `UserService`, there should be tests for `provisionUser`.